### PR TITLE
♻️Rework Auto Updater

### PR DESCRIPTION
**Pull request by Vyperus**
**Changes:**

1. Get rid of old update notifications.
2. Insert new update notification using the notification service.

PR: #1111

---

See the [examples](#example) for inspiration.

## How can others test the changes?

> Describe how others can test your changes (you can remove this section for typo fixes etc.)

## PR-Checklist

Please check the boxes in this list after submitting your PR:

- [x] You can merge this PR **right now** (if not, please prefix the title with "WIP: ")
- [x] I've tested **all** changes included in this PR.
- [x] I've also reviewed this PR myself before submitting (e.g. for scrambled letters, typos, etc.).
- [x] I've rebased the `develop` branch with my branch before finishing this PR.
- [x] I've **summarized all changes** in a list above.
- [x] I've mentioned all **PRs, which relate to this one**.
- [x] I've prefixed my Pull Request title is according to [gitmoji guide](https://gitmoji.carloscuesta.me/).

## Example

<details>
<summary>
:warning: Before merging please click here to expand and follow the guide.
</summary>
<br>

Please use `:twisted_rightwards_arrows:` at the beginning of your merge commit title.

Example 1:

<pre>
<code>
:twisted_rightwards_arrows: :bug: Fix Wrong Text Decoration at ...
</code>
</pre>

To get your commit message, just copy the first part of this pull request.

Example 2:
<pre>
<code>
**Changes:**

- Fixes Wrong Text Decoration at ...
- Fixes some typos
- ...

**Issues:**

Closes #NumberOfFixedIssue

PR: #NumberOfThisPR
</code>
</pre>
</details>

## Emojis

<details>
<summary>
Expand for a list of most used Emojis.
</summary>
<br>

Please prefix your commit messages with an Emoji.

Ref: https://gitmoji.carloscuesta.me/

| Description              | Glyphe               | Emoji  |
|--------------------------|----------------------|--------|
| Bugfix                   | `:bug:`              | 🐛     |
| Configuration releated   | `:wrench:`           | 🔧     |
| Cosmetic                 | `:lipstick:`         | 💄     |
| Dependencies Downgrade   | `:arrow_down:`       | ⬇️     |
| Dependencies Upgrade     | `:arrow_up:`         | ⬆️     |
| Formatting               | `:art:`              | 🎨     |
| Improving Performance    | `:zap:`              | ⚡️     |
| Initial commit           | `:tada:`             | 🎉     |
| Linter                   | `:rotating_light:`   | 🚨     |
| Miscellaneous            | `:package:`          | 📦     |
| New Feature              | `:sparkles:`         | ✨     |
| Refactoring Code         | `:recycle:`          | ♻️     |
| Releasing / Version tags | `:bookmark:`         | 🔖     |
| Removing Stuff           | `:fire:`             | 🔥     |
| Tests                    | `:white_check_mark:` | ✅     |
| Work In Progress (WIP)   | `:construction:`     | 🚧     |

</details>

### CODE REVIEW

File: src/main.ts [hunk 1]
```diff
@@ -108,5 +107,24 @@ export function configure(aurelia: Aurelia): void {
         notificationService.showNonDisappearingNotification(NotificationType.ERROR, errorMessage);
       });
     }
+
+    if ((window as any).nodeRequire) {
```
Comment (robpal94) [inline src/main.ts]: Extract business rule.

File: src/main.ts [hunk 2]
```diff
@@ -83,13 +83,13 @@ export function configure(aurelia: Aurelia): void {
   aurelia.start().then(() => {
     aurelia.setRoot();
 
+    const applicationRunsInElectron: boolean = (window as any).nodeRequire !== undefined;
     // check if the processengine started successfull
```
Comment (SteffenKn) [inline src/main.ts]: This comment is wrong and imo there is no comment needed
```suggestion
```

File: src/main.ts [hunk 3]
```diff
@@ -111,5 +111,27 @@ export function configure(aurelia: Aurelia): void {
...
+      });
+
+      ipcRenderer.on('update_available', (event: any) => {
+        notificationService.showNotification(NotificationType.INFO, 'Update available, started downloading.');
+      });
+
+      ipcRenderer.on('update_downloaded', (event: any) => {
+        const installButton: string = `<a href="javascript:nodeRequire('electron').ipcRenderer.send('quit_and_install')">here</a>`;
+
+        notificationService.showNonDisappearingNotification(NotificationType.INFO, `Update ready! Click ${installButton} to restart and install!`);
```
Comment (SteffenKn) [inline src/main.ts]: Maybe use `Update ready` as headline:

```suggestion
        const messageTitle: string = '<h4>Update ready!</h4>';
        const messageBody: string = `Click ${installButton} to restart and install!`;

        notificationService.showNonDisappearingNotification(NotificationType.INFO, `${messageTitle}\n${messageBody}`);
```

File: src/main.ts [hunk 4]
```diff
@@ -111,5 +111,27 @@ export function configure(aurelia: Aurelia): void {
...
+      const notificationService: NotificationService = aurelia.container.get('NotificationService');
+
+      ipcRenderer.send('app_ready');
+
+      ipcRenderer.on('update_error', (event: any) => {
+        notificationService.showNotification(NotificationType.INFO, 'Update Error');
+      });
+
+      ipcRenderer.on('update_available', (event: any) => {
+        notificationService.showNotification(NotificationType.INFO, 'Update available, started downloading.');
```
Comment (SteffenKn) [inline src/main.ts]:
```suggestion
        const messageTitle: string = '<h4>Update available.</h4>';
        const messageBody: string = `Download started.`;

        notificationService.showNonDisappearingNotification(NotificationType.INFO, `${messageTitle}\n${messageBody}`);
```

File: src/main.ts [hunk 5]
```diff
@@ -111,5 +110,33 @@ export function configure(aurelia: Aurelia): void {
...
       });
     }
+
+    if (applicationRunsInElectron) {
+      const ipcRenderer: any = (window as any).nodeRequire('electron').ipcRenderer;
+      const notificationService: NotificationService = aurelia.container.get('NotificationService');
+
+      ipcRenderer.send('app_ready');
+
+      ipcRenderer.on('update_error', (event: any) => {
```
Comment (alexanderkasten) [inline src/main.ts]: event is not used, please remove it

File: src/main.ts [hunk 6]
```diff
@@ -111,5 +110,33 @@ export function configure(aurelia: Aurelia): void {
...
+        notificationService.showNotification(NotificationType.INFO, 'Update Error');
+      });
+
+      ipcRenderer.on('update_available', (event: any) => {
```
Comment (alexanderkasten) [inline src/main.ts]: event is not used, please remove it

File: src/main.ts [hunk 7]
```diff
@@ -111,5 +110,33 @@ export function configure(aurelia: Aurelia): void {
...
+        const messageTitle: string = '<h4>Update available.</h4>';
+        const messageBody: string = `Download started.`;
+
+        notificationService.showNonDisappearingNotification(NotificationType.INFO, `${messageTitle}\n${messageBody}`);
+      });
+
+      ipcRenderer.on('update_downloaded', (event: any) => {
```
Comment (alexanderkasten) [inline src/main.ts]: event is not used, please remove it

File: src/main.ts [hunk 8]
```diff
@@ -111,5 +110,33 @@ export function configure(aurelia: Aurelia): void {
...
     }
+
+    if (applicationRunsInElectron) {
+      const ipcRenderer: any = (window as any).nodeRequire('electron').ipcRenderer;
+      const notificationService: NotificationService = aurelia.container.get('NotificationService');
+
+      ipcRenderer.send('app_ready');
+
+      ipcRenderer.on('update_error', (event: any) => {
+        notificationService.showNotification(NotificationType.INFO, 'Update Error');
```
Comment (alexanderkasten) [inline src/main.ts]:
```suggestion
        notificationService.showNotification(NotificationType.INFO, 'Update Error!');
```

**Pull request merged**